### PR TITLE
Feat/namespace

### PR DESCRIPTION
**Pull request by grvgoel81**
## Motivation and Context
* sessionNamespace param added in constructor.
* namespace param added in createSession and authorizeSession request params

Jira Link: https://toruslabs.atlassian.net/browse/PD-4265

## Description
* sessionNamespace param added in constructor.
* namespace param added in createSession and authorizeSession request params

## How has this been tested?
Runnung unit-test cases

## Screenshots (if appropriate):
Screenshot 2024-12-16 at 10 06 14 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project. (run lint)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My code requires a db migration.

### CODE REVIEW

File: Tests/SessionManagerTests/SessionManagementTest.swift [hunk 1]
```diff
@@ -17,15 +17,15 @@ final class SessionManagementTest: XCTestCase {
 
     func test_createSessionID() async throws {
         let sessionId = try SessionManager.generateRandomSessionID();
-        let session = SessionManager(sessionId: sessionId)
+        let session = SessionManager(sessionId: sessionId, sessionNamespace: "sfa")
```
Comment (metalurgical) [inline Tests/SessionManagerTests/SessionManagementTest.swift:20]: We'd need to test for the case in which sessionNamespace is nil here too.

Comment (grvgoel81) [reply Tests/SessionManagerTests/SessionManagementTest.swift:20]: Did that. Also passed without sessionNamespace and with sessionNamespace: nil.

**Review by metalurgical — APPROVED** [review]

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Comment by metalurgical** [comment]
@grvgoel81 We forgot about the podspec here for cocoapods. Please do a follow up PR to update it.

**Comment by grvgoel81** [comment]
> @grvgoel81 We forgot about the podspec here for cocoapods. Please do a follow up PR to update it.

Chai had pushed.
